### PR TITLE
[Scanner] Add persistence APIs for current execution

### PR DESCRIPTION
**Pull request by mkolodezny**

### CODE REVIEW

File: common/persistence/cassandra/cassandraPersistence.go [hunk 1]
```diff
@@ -440,6 +440,11 @@ workflow_state = ? ` +
 		`and visibility_ts = ? ` +
 		`and task_id = ?`
 
+	templateGetExecutionsQuery = `SELECT domain_id, workflow_id, run_id, current_run_id, workflow_state ` +
```
Comment (yux0) [inline common/persistence/cassandra/cassandraPersistence.go:443]: Can we reuse templateListWorkflowExecutionQuery?

Comment (mkolodezny) [reply common/persistence/cassandra/cassandraPersistence.go:443]: The templateListWorkflowExecutionQuery selects unnecessary stuff while we need to keep it as lite as possible for the scanner.

File: common/persistence/cassandra/cassandraPersistence.go [hunk 2]
```diff
@@ -2063,6 +2068,50 @@ func (d *cassandraPersistence) GetCurrentExecution(request *p.GetCurrentExecutio
 	}, nil
 }
 
+func (d *cassandraPersistence) ListCurrentExecutions(
+	request *p.ListCurrentExecutionsRequest,
+) (*p.ListCurrentExecutionsResponse, error) {
+	query := d.session.Query(
+		templateGetExecutionsQuery,
```
Comment (vancexu) [inline common/persistence/cassandra/cassandraPersistence.go:2075]: Maybe `templateListCurrentExecutionsQuery`?

File: common/persistence/cassandra/cassandraPersistence.go (hunk 1 above)
Comment (yux0) [reply common/persistence/cassandra/cassandraPersistence.go]: My personal preference is the simplicity over efficiency here as this triggers from a background job. But I don't have a strong opinion.

File: common/persistence/cassandra/cassandraPersistence.go [hunk 3]
```diff
@@ -2063,6 +2068,50 @@ func (d *cassandraPersistence) GetCurrentExecution(request *p.GetCurrentExecutio
...
+	for iter.MapScan(result) {
+		runID := result["run_id"].(gocql.UUID).String()
+		if runID != permanentRunID {
+			result = make(map[string]interface{})
+			continue
+		}
+		response.Executions = append(response.Executions, &p.CurrentWorkflowExecution{
+			DomainID:     result["domain_id"].(gocql.UUID).String(),
+			WorkflowID:   result["workflow_id"].(string),
+			RunID:        result["run_id"].(gocql.UUID).String(),
```
Comment (yux0) [inline common/persistence/cassandra/cassandraPersistence.go:2097]: The runID will always be permanentRunID. Not sure if you still want to return.

**Pull request merged**